bridge/test/nftbridge/NFTBridge_test.js

```diff
@@ -676,5 +692,157 @@ contract("Bridge NFT", async function(accounts) {
+        )
+        let anotherAccountBalance = await token.balanceOf(receiverAddress);
+        assert(
+            singleTokenBalance.eq(anotherAccountBalance), "Receiver account should have the token"
+        )
+      }
+
+      it("emits expected event and modifies affected balances correctly in successful case - throws error if re-claim is attempted",
+          async function () {
+            await assertTokenIsLockedInBridge(this.bridgeNft.address, anotherAccount, this.token);
+
+            const receipt = await this.sideBridgeNft.claim(
+                {
+                  to: anotherAccount, tokenId: tokenId, blockHash: blockHash, transactionHash: transactionHash,
+                  logIndex: logIndex
+                },
+                {from: anotherAccount}
+            );
+
+            truffleAssert.eventEmitted(receipt, claimedNFTTokenEventType, (event) => {
+              return (
+                  event._transactionHash === transactionHash &&
+                  event._originalTokenAddress === tokenAddress &&
+                  event._to === anotherAccount &&
+                  event._sender === anAccount &&
+                  event._tokenId.eq(new BN(tokenId)) &&
+                  event._blockHash === blockHash &&
+                  event._logIndex.eq(new BN(logIndex)) &&
+                  event._receiver === anotherAccount
+              );
+            });
+
+            let sideToken = await IERC721.at(sideTokenAddress);
+            await assertTokenHasBeenClaimed(this.sideBridgeNft.address, anotherAccount, sideToken);
+
+            await truffleAssert.fails(
+                this.sideBridgeNft.claim(
+                    {
+                      to: anotherAccount, tokenId: tokenId, blockHash: blockHash, transactionHash: transactionHash,
+                      logIndex: logIndex
+                    },
+                    {from: anotherAccount}
+                ),
+                "Bridge: Already claimed"
+            );
+
+            // Original token is still locked in main chain bridge, it was claimed (minted) in side chain.
+            await assertTokenIsLockedInBridge(this.bridgeNft.address, anotherAccount, this.token);
+            await assertTokenHasBeenClaimed(this.sideBridgeNft.address, anotherAccount, sideToken)
+          });
+
+      it("throws an error when transaction was not crossed", async function () {
+        await assertTokenIsLockedInBridge(this.bridgeNft.address, anotherAccount, this.token);
+
+        await truffleAssert.fails(
+            this.sideBridgeNft.claim(
+                {
+                  to: anotherAccount, tokenId: tokenId, blockHash: blockHash, transactionHash: utils.getRandomHash(),
+                  logIndex: logIndex
+                },
+                {from: anotherAccount}
+            ),
+            "Bridge: Tx not crossed"
+        );
+
+        await assertTokenIsLockedInBridge(this.bridgeNft.address, anotherAccount, this.token);

```

Comment:
If the transactions fails is not necessary  to check if the state is the same, as the blockchain ensures that. I know that the old bridge tests have a lot of examples like this, but you can omit that